### ISSUE TITLE
Further organization for saved sessions

### ISSUE THREAD

**Issue by Goshyiyo** [ReDEnergy/SessionSync#71]
This probably falls under more support for multi-window sessions. I skimmed through and couldn't find a suggestion 100% identical. I'll lay out a scenario. Let's say I have a saved session named "music" and have 100+ tabs inside this session and wish to separate them by genre. Tab groups or "windows" within the session. Similar to how the extension separates the windows (window 1, window 2, etc.) you have open in the active session tab. I'm not necessarily asking for multi-session windows on save (even though that'd help to serve this purpose), but a way for the user to manually create/delete "window" separators themselves inside saved sessions, and ideally drag/drop tabs inside/outside of them. Obviously, the ability to rename these "windows" just as you can rename tabs would be essential. A way to collapse the group would help keep things tidy and more usable in saved sessions with lots of tabs.

**Comment by ReDEnergy** [comment]
Thanks for the suggestion. This mostly covers everything in the nested session idea that I want to implement.. Basically to not be limited by the only 1 bookmark folder but to actually have at least 2 level of session saved so that you can do everything you just described (I do actually want to be able to move freely through nested sessions even though probably few users will use it but at least will bring a huge number of great features for those that do)
So... I can't really say that there's even 1 aspect of what you requested that I don't want to add.
I really want this myself a lot, but I can't really promise how fast will be implemented.

**Comment by Goshyiyo** [comment]
> 
> 
> Thanks for the suggestion. This mostly covers everything in the nested session idea that I want to implement.. Basically to not be limited by the only 1 bookmark folder but to actually have at least 2 level of session saved so that you can do everything you just described (I do actually want to be able to move freely through nested sessions even though probably few users will use it but at least will bring a huge number of great features for those that do)
> So... I can't really say that there's even 1 aspect of what you requested that I don't want to add.
> I really want this myself a lot, but I can't really promise how fast will be implemented sadly :(

Well, you put this much more simply than I did. Two levels. It's great to hear that this is something you want to implement in the future. Maybe having sessions collapsed under a main session would be more simple to program? It's less ideal to me personally, but serves the same purpose. I have no clue really. This is my only gripe with the addon, otherwise it's a complete dream and has everything I want. I'm just somebody that always has tons of tabs open for different projects and would like an option to separate different aspects of said projects. Thanks for the response.

**Comment by ReDEnergy** [comment]
No... actually that would be more difficult. I order to speed things up I can theoretically provide a temporal solution and enable nested walking through the bookmarks. I have to investigate it clearly but it might be the easiest solution to provide most of the features with a few more clicks, but I suppose it's better than nothing. At least until I can improve an improved UI as well.

**Comment by Goshyiyo** [comment]
Just echoing what you said, it's definitely better than nothing until you can get it done the way you want to.

**Comment by vpfaiz** [comment]
I was big fan of Session-buddy for Chrome but it lacks the sync feature. I guess the GUI and organizing of sessions is better there. May be you can adopt some ideas from there. Also see if you can suggest a title for a session from the titles of pages inside..

**Comment by anishmittal2020** [comment]
@ReDEnergy May be we could an ability to manually edit a session (When selected to edit manually, it could show the URL of all bookmarks in a session in an editor, which could be copied and pasted to and fromclipboard). Presently we can drag and drog each item one by one between sessions. This would enable to edit multiple bookmarks quickly.

**Comment by ReDEnergy** [comment]
@anishmittal2020 I know there are a lot of limitation including this one... Sometimes I get around using the Bookmark Manager. Just navigate to the folder and you can do a lot of stuff that the addon does not let you for now. Of course, everything you do to the bookmarks will reflect what the addon shows so as long as you know what you want to do everything is fine.
![image](https://user-images.githubusercontent.com/1395510/67530647-6cc12480-f6c8-11e9-8118-3cf32a82b5fb.png)

P.S As for the requested feature. I totally feel the need for this myself. It's just that I'm not really able to invest time in the addon right now. Some things are really challenging to modify and launch and just a few hours are not enough make the changes, I need a few days for complex things. Sadly, right know I can't really afford to do that because I'm in the middle of of important part of launching my indie game that I've been working for the past 2.5 years and I need to invest a serious amount of time to make sure everything is fine there. Once things will settle down I will definitely try to update the addon more often because I really care about this project and I kinda start to be annoyed myself that I have so many missing features that would be really useful.
